### ISSUE TITLE
~/Library/Caches/org.carthage.CarthageKit has weird permissions

### ISSUE THREAD

**Issue by kballard** [Carthage/Carthage#842]
I just checked out my `~/Library/Caches/org.carthage.CarthageKit` folder and it had weird permissions:

```
> ls -ld org.carthage.CarthageKit
d-wxrw--wt  4 kevinballard  staff  136 Oct 12 12:43 org.carthage.CarthageKit/
```

Note how it's not readable by the owner, but it is writable by the group and the world. The folders within have more sensible permissions:

```
drwxr-xr-x   3 kevinballard  staff  102 Oct 12 12:43 binaries/
drwxr-xr-x  14 kevinballard  staff  476 Oct 12 12:43 dependencies/
```

**Comment by webdevotion** [comment]
Same permission set for me:
```console
> ls -ld org.carthage.CarthageKit
d-wxrw--wt  3 webdevotion  staff  102 Oct  7 15:13 org.carthage.CarthageKit
```

I get permission denied when I try to `ls` inside the CarthageKit folder.  When I run `sudo ls` I only see `dependencies` listed.  No `binaries` folder for me.

```console
> sudo ls -ld
d-wxrw--wt  3 webdevotion  staff  102 Oct  7 15:13 .
> carthage version
0.9.3
> which carthage 
/usr/local/bin/carthage
```

FWIW: 
I ran into this possible issue while hunting down a similar problem 
as reported in issue #687  ( Strip exited with 1).

**Comment by sharplet** [comment]
I have a possibly related issue where whenever I run `carthage update`, the output is always `Cloning ...` instead of `Fetching ...`.

**Comment by sharplet** [comment]
And FWIW, the issue I mentioned appears to be resolved as of 0.9.4.

**Comment by Goles** [comment]
Hey, I just updated to `0.11.0` and I noticed that I have weird permissions on this folder too. That's actually preventing our CI pipeline to work properly.

This is what we're seeing:

```bash
d-wxrw--wt  4 _xcsbuildd  _xcs  136 Dec 14 12:03 org.carthage.CarthageKit/
```

As @webdevotion mentions, I get `permission denied` when I try to ls inside the CarthageKit folder. IMHO the permissions should have user level read, and read like:

```bash
drwxrw--wt  4 _xcsbuildd  _xcs  136 Dec 14 12:03 org.carthage.CarthageKit/
```

(note the `chmod u+r`)

Any ideas?
